lib/Dialect/FIRRTL/Transforms/GrandCentralTaps.cpp

```diff
@@ -709,11 +736,32 @@ void GrandCentralTapsPass::processAnnotation(AnnotatedPort &portAnno,
   auto targetAnno =
       targetAnnoIt != annos.end() ? targetAnnoIt->second : portAnno.anno;
 
+  // If the annotation is non-local, look up the corresponding NLA operation to
+  // find the exact instance path. We basically make the `wiring.prefices` array
+  // of instance paths list all the possible paths to the beginning of the NLA
+  // path. During wiring of the ports, we generate hierarchical names of the
+  // form `<prefix>.<nla-path>.<suffix>`. If we don't have an NLA, we leave it
+  // to the key-class-specific code below to come up with the possible prefices.
+  NonLocalAnchor nla;
+  if (auto nlaSym = targetAnno.getMember<FlatSymbolRefAttr>("circt.nonlocal")) {
+    nla = dyn_cast<NonLocalAnchor>(circuitSymbols->lookup(nlaSym.getAttr()));
+    assert(nla);
+    // Find all paths to the root of the NLA.
+    StringAttr rootSym =
+        TypeSwitch<Attribute, StringAttr>(nla.namepath()[0])
+            .Case<InnerRefAttr>([](auto attr) { return attr.getModule(); })
+            .Case<FlatSymbolRefAttr>([](auto attr) { return attr.getAttr(); });
+    Operation *root = circuitSymbols->lookup(rootSym);
```

Comment:
```suggestion
    Operation *root = circuitSymbols->lookup(nla.root());
```

FYI: there's a bunch of added member functions for extracting information from NLAs that help here: https://github.com/llvm/circt/blob/5b70f5bd7c0d7cb1ad9a8f073b74b6d1b995381b/include/circt/Dialect/FIRRTL/FIRRTLStructure.td#L178

Reply:
Cool stuff, thanks for pointing this out. Let me fix this up.